### PR TITLE
Platform 3807

### PR DESCRIPTION
**Pull request by ludwikkazmierczak**

### CODE REVIEW

File: maintenance/wikia/HttpsMigration/migrateWikiToFandom.php [hunk 1]
```diff
@@ -71,13 +71,22 @@ public function execute() {
 			if ( count( $data ) === 2 ) {
 				$targetDomain = $data[1];
 			} else {
+				# check if addictional /en alias domain should be created
+				global $additionalDomain;
```
Comment (macbre) [inline maintenance/wikia/HttpsMigration/migrateWikiToFandom.php]: Do we need a global here? Won't `$this->additionalDomain` be enough?

File: maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+				'topWikiArticles',
+				$wikiId,
+				Wikia\Search\TopWikiArticles::TOP_ARTICLES_CACHE,
+				true
+			),
+		];
+
+		foreach ( $keys as $key ) {
+			$wgMemc->delete( $key );
+		}
```
Comment (macbre) [inline maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php:90]: Why do we need to purge memcache entries? If I understand the script correctly, we only want to add an "alias" domain for English wikis. `foo.fandom.com/en` links were not (and won't be) generated for a given wiki, as MediaWiki will always use a base domain when generating URLs.

File: maintenance/wikia/HttpsMigration/migrateWikiToFandom.php [hunk 3]
```diff
@@ -116,8 +135,18 @@ private function getTargetDomain( $sourceDomain ) {
 			}
 			return "{$parts[1]}.{$wgFandomBaseDomain}/{$parts[0]}";
 		}
-
+		$additionalDomain = true;
```
Comment (macbre) [inline maintenance/wikia/HttpsMigration/migrateWikiToFandom.php]: Changing the state of an object inside a getter method is not really a "kosher" approach.

File: maintenance/wikia/HttpsMigration/migrateWikiToFandom.php [hunk 4]
```diff
@@ -115,7 +114,7 @@ public function execute() {
 			}
 
 			$this->output( "Wiki with ID {$sourceWikiId} was migrated from {$sourceDomain} to {$targetDomain}!\n" );
-			if ( $additionalDomain ){
+			if ( !stripos($targetDomain, "/")  ){
```
Comment (macbre) [inline maintenance/wikia/HttpsMigration/migrateWikiToFandom.php]: Do we need to be case insensitive here? I'm not familiar with upper slash? ;)

**Review by macbre — APPROVED** [review]

File: maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ * Script to create /en alias for migrated English wikis
+ * Input: CSV file with list of wiki IDs, one per line
+ * e.g.
+ * 	ghostbusters.fandom.com will have additional /en alias domain: ghostbusters.fandom.com/en
+ * 	for ar.ghostbusters.fandom.com additional domain with /en won't be created
+ *
+ * The script runs in dry mode by default, and the --saveChanges option
+ * needs to be used when you actually want to save the changes.
+ */
+class MigrateWikiToFandom extends Maintenance {
```
Comment (rogatty) [inline maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php]:
```suggestion
class CreateEnAliasForMigratedWikis extends Maintenance {
```

File: maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+		if ( $parts[1] !== 'fandom' ){
+			$this->output( "Failed to create /en alias for {$sourceDomain}!\n" );
+			return false;
+		}
+		return "{$parts[0]}.{$wgFandomBaseDomain}/en";
+	}
+
+}
+
+$maintClass = 'MigrateWikiToFandom';
```
Comment (rogatty) [inline maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php]:
```suggestion
$maintClass = 'CreateEnAliasForMigratedWikis';
```

File: maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+			$additionalTargetDomain = $this->getAdditionalDomain( $sourceDomain );
+			if ( !$additionalTargetDomain ) {
+				continue;
+			}
+
+
+			if ( $saveChanges ) {
+
+				if ( !empty($additionalTargetDomain) ){
```
Comment (rogatty) [inline maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php]: We already check it in line 59.

File: maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+					WikiFactory::addDomain( $sourceWikiId, $additionalTargetDomain, 'Creating /en alias for Wiki with ID {$sourceWikiId}' );
+				}
+			}
+
+			if ( !empty($additionalTargetDomain)  ){
```
Comment (rogatty) [inline maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php]: We already check it in line 59.

File: maintenance/wikia/HttpsMigration/migrateWikiToFandom.php [hunk 9]
```diff
@@ -86,10 +94,15 @@ public function execute() {
 			}
 
 			if ( $saveChanges ) {
+
 				WikiFactory::addDomain( $sourceWikiId, $targetDomain, 'Migration to fandom.com' );
 
 				WikiFactory::setmainDomain( $sourceWikiId, $targetDomain, 'Migration to fandom.com' );
 
+				if ( strpos($targetDomain, "/") === false  ){
```
Comment (rogatty) [inline maintenance/wikia/HttpsMigration/migrateWikiToFandom.php]:
```suggestion
				if ( !empty( $additionalTargetDomain )  ){
```

File: maintenance/wikia/HttpsMigration/migrateWikiToFandom.php [hunk 10]
```diff
@@ -101,13 +114,17 @@ public function execute() {
 			}
 
 			$this->output( "Wiki with ID {$sourceWikiId} was migrated from {$sourceDomain} to {$targetDomain}!\n" );
+			if ( strpos($targetDomain, "/") === false ){
```
Comment (rogatty) [inline maintenance/wikia/HttpsMigration/migrateWikiToFandom.php]:
```suggestion
			if ( !empty( $additionalTargetDomain )  ){
```
or just move it below the actual action.

File: maintenance/wikia/HttpsMigration/migrateWikiToFandom.php [hunk 11]
```diff
@@ -116,8 +133,17 @@ private function getTargetDomain( $sourceDomain ) {
 			}
 			return "{$parts[1]}.{$wgFandomBaseDomain}/{$parts[0]}";
 		}
-
 		return "{$parts[0]}.{$wgFandomBaseDomain}";
+
+	}
+
+	private function getAdditionalDomain( $sourceDomain ) {
```
Comment (rogatty) [inline maintenance/wikia/HttpsMigration/migrateWikiToFandom.php]: This name is unclear, how about `getEnglishAliasDomain`?

**Review by rogatty — APPROVED** [review]

**Pull request merged**